prov/rxm/src/rxm.h

```diff
@@ -504,6 +513,42 @@ void rxm_ep_progress_multi(struct util_ep *util_ep);
 int rxm_ep_prepost_buf(struct rxm_ep *rxm_ep, struct fid_ep *msg_ep);
 int rxm_send_queue_init(struct rxm_ep *rxm_ep, struct rxm_send_queue **send_queue, size_t size);
 void rxm_send_queue_close(struct rxm_send_queue *send_queue);
+void rxm_ep_progress_deferred_queues(struct rxm_ep *rxm_ep);
+
+static inline void
+rxm_ep_enqueue_deferred_tx_queue(struct rxm_tx_entry *tx_entry)
+{
+	if (dlist_empty(&tx_entry->deferred_tx_entry)) {
+		dlist_insert_tail(&tx_entry->conn->deferred_tx_queue,
+				  &tx_entry->deferred_tx_entry);
+		if (dlist_empty(&tx_entry->conn->deferred_conn_entry))
+			dlist_insert_tail(&tx_entry->ep->deferred_conn_queue,
+					  &tx_entry->conn->deferred_conn_entry);
+	}
+}
```

Comment:
Only queuing when the list is empty is non-intuitive.  Why doesn't this always just queue?

Reply:
Sorry, I think I need to add more comments why it so.
This is not the list, this is the entry of the list. We just check whether the entry is already queued or not. If not - insert to the queue, yes - skip this operations.

Reply:
We should really know if the tx_entry has been queued or not.  Why are checks needed?  The conn_queue makes more sense to me.  But I would rather see something like this:
```
if (dlist_empty(conn->deferred_queue))
    dlist_insert_tail(ep->deferred_queue, conn->deferred_entry);
dlist_insert_tail(conn->deferred_queue, tx_entry->entry);
```
This makes it clearer that the connection's deferred queue state is based on it being empty and aligns it better with the removal function below.

Reply:
Yes, this makes sense! Done. Thanks!